motion_computation/CMakeLists.txt

```diff
@@ -1,90 +1,108 @@
+# You may obtain a copy of the License at
 #
-# http://www.apache.org/licenses/LICENSE-2.0
+#     http://www.apache.org/licenses/LICENSE-2.0
 #
 # Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
-# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
-# License for the specific language governing permissions and limitations under
-# the License.
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
 
 cmake_minimum_required(VERSION 3.5)
 project(motion_computation)
 
-# Declare carma package and check ROS version
+# CARMA builds packages in an environment with both ROS 1 and ROS 2 installed.
+# This check gracefully skips the current package if the sourced ROS environment
+# is not the specified version. This call must come before any other ROS
+# dependencies becasue ROS 1 does not have some of the required packages.
 find_package(carma_cmake_common REQUIRED)
 carma_check_ros_version(2)
-carma_package()
 
-## Find dependencies using ament auto
+
+# Added outside of `dependencies.cmake` because ament sets some variables
+# (e.g., BUILD_TESTING) that affect the configuration options for the rest of
+# the package. Putting the command call here allows us to put all project
+# options together in separate CMake module then query those options in
+# `dependencies.cmake`.
 find_package(ament_cmake_auto REQUIRED)
-ament_auto_find_build_dependencies()
 
-# Name build targets
-set(worker_lib motion_computation_worker_lib)
-set(node_exec motion_computation_node_exec)
-set(node_lib motion_computation_node_lib)
+include(options.cmake)
+include(dependencies.cmake)
 
-# Includes
-include_directories(
-  include
-)
+# The generated compilation database is helpful with code completion in IDEs
+set(CMAKE_EXPORT_COMPILE_COMMANDS ${motion_computation_EXPORT_COMPILE_COMMANDS})
 
-# boost::posix_time definition for using nanoseconds
-add_definitions(-DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG)
+# This prevents `colcon` from tyring to build the CMake project's binary
+# directory. This is useful in case we want to build the project outside of
+# `colcon` but still keep it in a ROS repository.
+file(TOUCH ${PROJECT_BINARY_DIR}/COLCON_IGNORE)
 
-# Build
+# Configures CARMA package default settings
+carma_package()
 
-ament_auto_add_library(${worker_lib} SHARED
-        src/motion_computation_worker.cpp
-        src/mobility_path_to_external_object.cpp
-        src/psm_to_external_object_convertor.cpp
-        src/bsm_to_external_object_convertor.cpp
+# C17 CMake support added in CMake 3.21
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_C_STANDARD 11)
+
+# ament_auto will automatically add include files from include/roadway_objects
```

Comment:
Are you suggesting `# ament_auto` be changed to `# motion_computation`? If so, I think it should remain. My intent was to let readers know that `ament_auto` will automatically include the necessary files since its not clear from the CMake function's name. You have to already know what the function does. Thoughts?

Reply:
sorry I meant there was a typo, it is not for roadway_objects, but motion_computation.

Reply:
Oh I see. Thanks.